Assets/HoloToolkit/Input/Scripts/Interactable.cs

```diff
@@ -0,0 +1,100 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using UnityEngine;
+
+namespace HoloToolkit.Unity
+{
+    /// <summary>
+    /// Base class Interactable GameObjects can inherit from.
+    /// </summary>
+    public class Interactable : MonoBehaviour
+    {
+        protected bool TriggerKeywordOnGazeOnly;
+
+        protected bool IsFocused;
+
+        protected virtual void OnEnable()
+        {
+            GestureManager.Instance.OnTap += OnTap;
+            GazeManager.Instance.OnGazeExit += OnGazeExit;
+            GazeManager.Instance.OnGazeEnter += OnGazeEnter;
+            KeywordManager.Instance.OnKeywordRecognized += KeywordRecognized;
+        }
+
+        protected virtual void OnDisable()
+        {
+            GestureManager.Instance.OnTap -= OnTap;
+            GazeManager.Instance.OnGazeExit -= OnGazeExit;
+            GazeManager.Instance.OnGazeEnter -= OnGazeEnter;
+            KeywordManager.Instance.OnKeywordRecognized -= KeywordRecognized;
+        }
+
+        /// <summary>
+        /// Called when a user has tapped any gameObject.
+        /// </summary>
+        /// <param name="tappedGameObject">GameObject user has tapped.</param>
+        protected virtual void OnTap(GameObject tappedGameObject)
+        {
+            if (tappedGameObject == gameObject)
```

Comment:
I'd like to propose that we switch from events/delegates to an interface so we can stop digging in.  I feel like I've generated enough data to have confidence that the cost of the interface is at worst the same perf as send message and provides the code readability benefits of an event.  I also feel like I've generated enough data to show that the event pattern will be overall more expensive on the CPU for most applications.  

If you continue to dig to try to find data that supports the event handler, do be sure to have your event handler be something like an Action<GameObject> and have the event handler compare the passed in GameObject to itself and take an action (like incrementing a value that you print out and the end of the program) if they do match.  Also be sure to measure the total cost of raising the event, not just the time spent for one of the event handlers to receive the message.